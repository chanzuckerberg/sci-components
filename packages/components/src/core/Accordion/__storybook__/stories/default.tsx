import { Args } from "@storybook/react";
import RawAccordion, {
  AccordionDetails,
  AccordionHeader,
} from "src/core/Accordion";
import { EXTRA_SHORT_LOREM_IPSUM } from "src/common/storybook/loremIpsum";

export const Accordion = (props: Args): JSX.Element => {
  const { id, subtitle, useDivider, togglePosition } = props;
  return (
    <RawAccordion
      id={id}
      useDivider={useDivider}
      togglePosition={togglePosition}
      {...props}
    >
      <AccordionHeader id={`${id}-header`} subtitle={subtitle}>
        Accordion Header
      </AccordionHeader>
      <AccordionDetails>{EXTRA_SHORT_LOREM_IPSUM}</AccordionDetails>
    </RawAccordion>
  );
};
