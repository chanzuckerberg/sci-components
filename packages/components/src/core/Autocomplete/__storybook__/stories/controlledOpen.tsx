import { Args } from "@storybook/types";
import { DefaultAutocompleteOption } from "src/core/Autocomplete/components/AutocompleteBase";
import { AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS } from "src/common/storybook/AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS";
import { useState } from "react";
import RawAutocomplete from "src/core/Autocomplete";
import { AUTOCOMPLETE_WRAPPER_STYLES } from "../constants";

export const ControlledOpenDemo = <T extends DefaultAutocompleteOption>(
  props: Args
): JSX.Element => {
  const {
    label,
    multiple,
    options = AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS,
  } = props;

  const [open, setOpen] = useState(true);

  return (
    <div style={AUTOCOMPLETE_WRAPPER_STYLES}>
      <RawAutocomplete
        onClickAway={() => {
          setOpen(false);
        }}
        onClick={() => {
          if (open) {
            setOpen(false);
          } else {
            setOpen(true);
          }
        }}
        open={open}
        id="autocomplete-demo"
        disableCloseOnSelect={multiple}
        label={label}
        multiple={multiple}
        options={options}
        getOptionDisabled={(option: T) => {
          return option.name === "Type: feature request";
        }}
        {...props}
      />
    </div>
  );
};
