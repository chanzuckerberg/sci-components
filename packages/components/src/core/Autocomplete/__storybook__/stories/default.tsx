import { Args } from "@storybook/react";
import { DefaultAutocompleteOption } from "src/core/Autocomplete/components/AutocompleteBase";
import { AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS } from "src/common/storybook/AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS";
import { SyntheticEvent, useEffect, useState } from "react";
import RawAutocomplete, {
  AutocompleteMultiColumnValue,
  SDSAutocompleteValue,
} from "src/core/Autocomplete";
import TagFilter from "src/core/TagFilter";
import { AutocompleteValue } from "@mui/base";
import {
  AUTOCOMPLETE_DIV_MARGIN,
  AUTOCOMPLETE_WRAPPER_STYLES,
} from "../constants";

export const Autocomplete = <
  T extends DefaultAutocompleteOption,
  Multiple extends boolean | undefined,
>(
  props: Args
): JSX.Element => {
  const {
    label,
    multiple,
    options = AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS,
    search,
    value: propValue,
    keepSearchOnSelect,
  } = props;

  type DisableClearable = false;
  type FreeSolo = false;

  const isControlled = propValue !== undefined;
  const [value, setValue] = useState<
    SDSAutocompleteValue<T, Multiple, DisableClearable, FreeSolo>
  >(
    (multiple ? [] : null) as SDSAutocompleteValue<
      T,
      Multiple,
      DisableClearable,
      FreeSolo
    >
  );

  const [selection, setSelection] = useState<string[]>([]);

  useEffect(() => {
    if (isControlled) {
      setValue(propValue);
    }
  }, [isControlled, propValue]);

  useEffect(() => {
    setSelection([]);
    setValue(
      (multiple ? [] : null) as SDSAutocompleteValue<
        T,
        Multiple,
        DisableClearable,
        FreeSolo
      >
    );
  }, [multiple, options]);

  return (
    <>
      <div style={AUTOCOMPLETE_WRAPPER_STYLES}>
        <RawAutocomplete
          id="autocomplete-demo"
          disableCloseOnSelect={multiple}
          label={label}
          multiple={multiple}
          onChange={handleChange}
          keepSearchOnSelect={keepSearchOnSelect}
          options={options}
          search={search}
          value={value}
          getOptionDisabled={(option: T) => {
            return option.name === "Type: feature request";
          }}
          {...props}
        />
      </div>
      <div style={{ margin: AUTOCOMPLETE_DIV_MARGIN }}>
        {selection.length
          ? selection.map((item) => {
              return (
                <TagFilter
                  key={item}
                  label={item}
                  onDelete={() => handleTagDelete(item)}
                  onClick={() => handleTagDelete(item)}
                />
              );
            })
          : null}
      </div>
    </>
  );

  // eslint-disable-next-line sonarjs/cognitive-complexity
  function handleChange(
    _: SyntheticEvent<Element, Event>,
    newValue: SDSAutocompleteValue<T, Multiple, DisableClearable, FreeSolo>
  ) {
    setValue(newValue);
    const newSelection: string[] = [];

    if (
      (newValue as T)?.name ||
      (Array.isArray(newValue) && (newValue as T[])[0].name)
    ) {
      if (multiple) {
        setSelection(
          Array.isArray(newValue) ? newValue?.map((item) => item.name) : []
        );
      } else {
        if (newValue && !Array.isArray(newValue) && newValue.name) {
          setSelection([(newValue as T).name]);
        }
      }
    } else {
      if (multiple) {
        if (newValue) {
          Object.values(newValue).forEach((items) => {
            (items as T[])?.map(({ name }: { name: string }) => {
              newSelection.push(name);
            });
          });
          setSelection(newSelection);
        }
      } else {
        if (newValue) {
          Object.values(newValue).forEach((item) => {
            newSelection.push((item as T).name);
          });
        }
        setSelection(newSelection);
      }
    }
  }

  // eslint-disable-next-line sonarjs/cognitive-complexity
  function handleTagDeleteMultiColumn(tag: string) {
    if (multiple) {
      if (value) {
        Object.entries(value).forEach(([key, values]) => {
          const index = (values as T[])?.findIndex(
            (item: T) => item.name === tag
          );
          if (index > -1) {
            const newValue = [...(values as T[])];
            newValue.splice(index, 1);
            setValue((prev) => {
              return {
                ...prev,
                [key]: newValue as AutocompleteValue<
                  T,
                  Multiple,
                  DisableClearable,
                  FreeSolo
                >,
              };
            });

            const newSelection = [...selection];
            const deleteIndex = newSelection.indexOf(tag);
            newSelection.splice(deleteIndex, 1);
            setSelection(newSelection);
          }
        });
      }
    } else {
      if (value) {
        Object.entries(value).forEach(([key, val]) => {
          if ((val as T).name === tag) {
            const finalValue = { ...value } as AutocompleteMultiColumnValue<
              T,
              Multiple,
              DisableClearable,
              FreeSolo
            >;
            if (finalValue) delete finalValue[key];
            setValue(finalValue);

            const newSelection = [...selection];
            const deleteIndex = newSelection.indexOf(tag);
            newSelection.splice(deleteIndex, 1);
            setSelection(newSelection);
          }
        });
      }
    }
  }

  function handleTagDelete(tag: string) {
    if (
      (value && (value as T).name) ||
      (Array.isArray(value) && (value as T[])[0].name)
    ) {
      if (multiple && Array.isArray(value)) {
        const index = value?.findIndex((item) => item.name === tag);
        const newValue = [...value];
        newValue.splice(index, 1);
        setValue(
          newValue as SDSAutocompleteValue<
            T,
            Multiple,
            DisableClearable,
            FreeSolo
          >
        );

        const newSelection = [...selection];
        const deleteIndex = newSelection.indexOf(tag);
        newSelection.splice(deleteIndex, 1);
        setSelection(newSelection);
      } else {
        setValue(
          null as SDSAutocompleteValue<T, Multiple, DisableClearable, FreeSolo>
        );
        setSelection([]);
      }
    } else {
      handleTagDeleteMultiColumn(tag);
    }
  }
};
