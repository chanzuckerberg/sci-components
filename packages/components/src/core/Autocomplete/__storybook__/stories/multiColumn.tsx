import { Args } from "@storybook/react";
import { DefaultAutocompleteOption } from "src/core/Autocomplete/components/AutocompleteBase";
import { AUTOCOMPLETE_MULTI_COLUMN_OPTIONS } from "src/common/storybook/AUTOCOMPLETE_MULTI_COLUMN_OPTIONS";
import { useEffect, useState } from "react";
import { AutocompleteMultiColumnValue } from "src/core/Autocomplete";
import {
  AutocompleteChangeDetails,
  AutocompleteChangeReason,
  AutocompleteValue,
} from "@mui/base";
import {
  AUTOCOMPLETE_DIV_MARGIN,
  AUTOCOMPLETE_WRAPPER_STYLES,
} from "../constants";
import TagFilter from "src/core/TagFilter";
import RawAutocompleteMultiColumn from "src/core/Autocomplete/components/AutocompleteMultiColumn";

export const AutocompleteMultiColumnDemo = <
  T extends DefaultAutocompleteOption,
  Multiple extends boolean | undefined,
>(
  props: Args
): JSX.Element => {
  const {
    multiple,
    options = AUTOCOMPLETE_MULTI_COLUMN_OPTIONS,
    search,
    label,
  } = props;

  type DisableClearable = false;
  type FreeSolo = false;

  const [selection, setSelection] = useState<string[]>([]);

  const [theValue, setTheValue] =
    useState<
      AutocompleteMultiColumnValue<T, Multiple, DisableClearable, FreeSolo>
    >();

  useEffect(() => {
    setSelection([]);
  }, [multiple]);

  const handleChange = (
    _event: React.SyntheticEvent,
    newValue: AutocompleteMultiColumnValue<
      T,
      Multiple,
      DisableClearable,
      FreeSolo
    >,
    _reason: AutocompleteChangeReason,
    _details?: AutocompleteChangeDetails<T>
  ) => {
    setTheValue(newValue);
    const newSelection: string[] = [];

    if (multiple) {
      if (newValue) {
        Object.values(newValue).forEach((items) => {
          (items as T[])?.map(({ name }: { name: string }) => {
            newSelection.push(name);
          });
        });
        setSelection(newSelection);
      }
    } else {
      if (newValue) {
        Object.values(newValue).forEach((item) => {
          newSelection.push((item as T).name);
        });
      }
      setSelection(newSelection);
    }
  };

  return (
    <>
      <div style={AUTOCOMPLETE_WRAPPER_STYLES}>
        <RawAutocompleteMultiColumn<T, Multiple, DisableClearable, FreeSolo>
          label={label}
          disableCloseOnSelect={false}
          multiple={multiple}
          onChange={handleChange}
          options={options}
          search={search}
          value={theValue}
          {...props}
        />
      </div>
      <div style={{ margin: AUTOCOMPLETE_DIV_MARGIN }}>
        {selection.length
          ? selection.map((item) => {
              return (
                <TagFilter
                  key={item}
                  label={item}
                  onDelete={() => handleTagDelete(item)}
                  onClick={() => handleTagDelete(item)}
                />
              );
            })
          : null}
      </div>
    </>
  );

  function handleTagDelete(tag: string) {
    if (multiple) {
      if (theValue) {
        Object.entries(theValue).forEach(([key, values]) => {
          const index = (values as T[])?.findIndex(
            (item: T) => item.name === tag
          );
          if (index > -1) {
            const newValue = [...(values as T[])];
            newValue.splice(index, 1);
            setTheValue((prev) => {
              return {
                ...prev,
                [key]: newValue as AutocompleteValue<
                  T,
                  Multiple,
                  DisableClearable,
                  FreeSolo
                >,
              };
            });

            const newSelection = [...selection];
            const deleteIndex = newSelection.indexOf(tag);
            newSelection.splice(deleteIndex, 1);
            setSelection(newSelection);
          }
        });
      }
    } else {
      if (theValue) {
        Object.entries(theValue).forEach(([key, value]) => {
          if ((value as T).name === tag) {
            const finalValue = { ...theValue };
            delete finalValue[key];
            setTheValue(finalValue);

            const newSelection = [...selection];
            const deleteIndex = newSelection.indexOf(tag);
            newSelection.splice(deleteIndex, 1);
            setSelection(newSelection);
          }
        });
      }
    }
  }
};
