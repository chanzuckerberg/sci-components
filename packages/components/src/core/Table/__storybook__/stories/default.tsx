import { RadioGroup } from "@mui/material";
import { Args } from "@storybook/react";
import CellBasic from "src/core/CellBasic";
import CellComponent from "src/core/CellComponent";
import CellHeader from "src/core/CellHeader";
import Icon from "src/core/Icon";
import InputRadio from "src/core/InputRadio";
import TableHeader from "src/core/TableHeader";
import TableRow from "src/core/TableRow";
import Tag from "src/core/Tag";
import { SdsTagColorType } from "src/core/Tag/style";
import RawTable from "src/core/Table";
import { StyledIconCell } from "../style";

export const Table = (props: Args): JSX.Element => {
  return (
    <RawTable {...props}>
      <TableHeader>
        <CellHeader horizontalAlign="center" hideSortIcon>
          Category
        </CellHeader>
        <CellHeader active>Active Header</CellHeader>
        <CellHeader>
          A very long table header title to test sort icon positioning
        </CellHeader>
        <CellHeader hideSortIcon>Component</CellHeader>
        <CellHeader horizontalAlign="right" hideSortIcon>
          Right Aligned and Not sortable
        </CellHeader>
      </TableHeader>
      <tbody>
        <TableRow>
          <CellComponent verticalAlign="center" horizontalAlign="center">
            <StyledIconCell>
              <Icon sdsSize="xl" sdsIcon="Flask" sdsType="static" />
              <Tag
                color="info"
                label="Chemistry"
                sdsStyle="rounded"
                sdsType="secondary"
                hover={false}
              />
            </StyledIconCell>
          </CellComponent>
          <CellBasic
            primaryText="Primary Text"
            secondaryText="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
            secondaryTextWrapLineCount={2}
            shouldTextWrap
          />
          <CellBasic
            primaryText="Primary Text"
            secondaryText="Secondary Text"
            tertiaryText="Tertiary Text"
            shouldShowTooltipOnHover={false}
          />
          <CellComponent>
            {["info", "notice", "beta", "negative", "positive"].map((item) => (
              <Tag
                key={item}
                color={item as SdsTagColorType}
                label={item as string}
                sdsStyle="rounded"
                sdsType="secondary"
                hover={false}
              />
            ))}
          </CellComponent>
          <CellBasic
            primaryText="356"
            horizontalAlign="right"
            shouldShowTooltipOnHover={false}
          />
        </TableRow>

        <TableRow>
          <CellComponent verticalAlign="center" horizontalAlign="center">
            <StyledIconCell>
              <Icon sdsSize="xl" sdsIcon="Download" sdsType="static" />
              <Tag
                color="info"
                label="Downloadable Content"
                sdsStyle="rounded"
                sdsType="secondary"
                hover={false}
              />
            </StyledIconCell>
          </CellComponent>
          <CellBasic
            primaryText="Primary Text"
            secondaryText="Secondary Text"
            tertiaryText="Tertiary Text"
            shouldShowTooltipOnHover={false}
          />
          <CellBasic
            primaryText="Primary Text"
            secondaryText="Secondary Text"
            shouldShowTooltipOnHover={false}
          />
          <CellComponent>
            <RadioGroup
              aria-labelledby="demo-radio-buttons-group-label"
              name="radio-buttons-group"
              defaultValue="1"
            >
              <InputRadio label="Option I" caption="Caption I" value="1" />
              <InputRadio label="Option II" caption="Caption II" value="2" />
            </RadioGroup>
          </CellComponent>
          <CellBasic
            primaryText="1,234"
            secondaryText="2,344,000"
            tertiaryText="12.5%"
            horizontalAlign="right"
            shouldShowTooltipOnHover={false}
          />
        </TableRow>

        <TableRow
          disabled
          shouldShowTooltipOnHover
          tooltipText="This row is DISABLED!"
          tooltipSubtitle="Tooltip subtitle"
        >
          <CellComponent verticalAlign="center" horizontalAlign="center">
            <StyledIconCell>
              <Icon sdsSize="xl" sdsIcon="Bacteria" sdsType="static" />
              <Tag
                color="negative"
                label="Disease"
                sdsStyle="rounded"
                sdsType="secondary"
                hover={false}
              />
            </StyledIconCell>
          </CellComponent>
          <CellBasic
            primaryText="Primary Text"
            secondaryText="Secondary Text"
            tertiaryText="Tertiary Text"
            shouldShowTooltipOnHover={false}
            verticalAlign="center"
          />
          <CellBasic
            primaryText="Primary Text"
            shouldShowTooltipOnHover={false}
            verticalAlign="center"
          />
          <CellComponent verticalAlign="center">
            <Icon sdsSize="l" sdsIcon="Virus" sdsType="static" />
          </CellComponent>
          <CellBasic
            primaryText="0.4"
            horizontalAlign="right"
            verticalAlign="center"
            shouldShowTooltipOnHover={false}
          />
        </TableRow>
      </tbody>
    </RawTable>
  );
};
