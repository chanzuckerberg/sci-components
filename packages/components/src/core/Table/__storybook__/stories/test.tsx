import CellBasic from "src/core/CellBasic";
import RawTable from "src/core/Table";
import TableRow from "src/core/TableRow";

export const TestDemo = (): JSX.Element => (
  <RawTable data-testid="Table">
    <tbody>
      <TableRow>
        <CellBasic primaryText="Primary 1" shouldShowTooltipOnHover={false} />
        <CellBasic primaryText="Primary 2" shouldShowTooltipOnHover={false} />
        <CellBasic primaryText="Primary 3" shouldShowTooltipOnHover={false} />
        <CellBasic primaryText="Primary 4" shouldShowTooltipOnHover={false} />
      </TableRow>
    </tbody>
  </RawTable>
);
