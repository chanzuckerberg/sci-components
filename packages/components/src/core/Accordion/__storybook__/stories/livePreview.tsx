import { Args } from "@storybook/react";
import { Accordion } from "./default";

const livePreviewStyles = {
  display: "grid",
  gridColumnGap: "24px",
  gridRowGap: "50px",
  gridTemplateColumns: "repeat(4, 200px)",
};

export function LivePreviewDemo(props: Args): JSX.Element {
  return (
    <div style={livePreviewStyles as React.CSSProperties}>
      <Accordion
        id="accordion-1"
        togglePosition="right"
        useDivider
        {...props}
      />
      <Accordion
        id="accordion-2"
        togglePosition="right"
        subtitle="Optional Subtitle"
        useDivider
        {...props}
      />
      <Accordion id="accordion-3" togglePosition="left" useDivider {...props} />
      <Accordion
        id="accordion-4"
        togglePosition="left"
        subtitle="Optional Subtitle"
        useDivider
        {...props}
      />
    </div>
  );
}
