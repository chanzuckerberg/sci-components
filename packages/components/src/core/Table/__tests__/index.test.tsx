import { generateSnapshots } from "@chanzuckerberg/story-utils";
import { composeStories } from "@storybook/react";
import { render, screen } from "@testing-library/react";
import * as stories from "../__storybook__/index.stories";

// Returns a component that already contain all decorators from story level, meta level and global level.
const { Test } = composeStories(stories);

describe("<Table />", () => {
  generateSnapshots(stories);

  it("renders table component", () => {
    render(<Test />);
    const elements = screen.getAllByTestId("Table");
    expect(elements).toBeTruthy();
  });

  it("renders table component children", async () => {
    render(<Test />);
    await screen.findByText("Primary 1");
  });
});
