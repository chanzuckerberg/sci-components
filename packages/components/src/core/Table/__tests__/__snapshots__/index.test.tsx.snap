// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> Default story renders snapshot 1`] = `
<table
  class="css-mpwjyz"
>
  <thead
    class="css-1426k2b"
  >
    <tr
      class="css-2l3m6x"
    >
      <th
        class="css-19ju8um"
      >
        <div
          class="css-187gjam"
        >
          <span>
            Category
          </span>
        </div>
      </th>
      <th
        class="css-5bfdy4"
      >
        <div
          class="css-rm9mvq"
        >
          <span>
            Active Header
          </span>
          <div
            class="css-1ft6wgl"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-tlyfhc-MuiSvgIcon-root"
              data-file-name="IconChevronDownSmall"
              data-testid="IconChevronDownSmall"
              fillcontrast="white"
              focusable="false"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            />
          </div>
        </div>
      </th>
      <th
        class="css-5bfdy4"
      >
        <div
          class="css-rm9mvq"
        >
          <span>
            A very long table header title to test sort icon positioning
          </span>
          <div
            class="css-1ft6wgl"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-12jh8eh-MuiSvgIcon-root"
              data-file-name="IconChevronDownSmall"
              data-testid="IconChevronDownSmall"
              fillcontrast="white"
              focusable="false"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            />
          </div>
        </div>
      </th>
      <th
        class="css-5bfdy4"
      >
        <div
          class="css-rm9mvq"
        >
          <span>
            Component
          </span>
        </div>
      </th>
      <th
        class="css-vnxybn"
      >
        <div
          class="css-wg04tn"
        >
          <span>
            Right Aligned and Not sortable
          </span>
        </div>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      class="css-1lvje81"
    >
      <td
        class="cell-component css-4zu111"
      >
        <div
          class="css-9lvjku"
        >
          <div
            class="css-1ft6wgl"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-25m3b5-MuiSvgIcon-root"
              data-file-name="IconFlaskLarge"
              data-testid="IconFlaskLarge"
              fillcontrast="white"
              focusable="false"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            />
          </div>
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-elc81v-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Chemistry
            </span>
          </div>
        </div>
      </td>
      <td
        class=" css-p6nomj"
        data-mui-internal-clone-element="true"
      >
        <div
          class="css-ov1ktg"
        >
          <div
            class="css-1ago99h"
          >
            <div
              class="css-7m17a9"
            >
              <div
                class="css-1asg04x"
              >
                Primary Text
              </div>
            </div>
            <span
              class="css-102yezv"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            </span>
          </div>
        </div>
      </td>
      <td
        class="css-p6nomj"
      >
        <div
          class="css-ov1ktg"
        >
          <div
            class="css-1ago99h"
          >
            <div
              class="css-7m17a9"
            >
              <div
                class="css-1asg04x"
              >
                Primary Text
              </div>
            </div>
            <span
              class="css-1xqvcqn"
            >
              Secondary Text
            </span>
            <span
              class="css-1xqvcqn"
            >
              Tertiary Text
            </span>
          </div>
        </div>
      </td>
      <td
        class="cell-component css-4umyx6"
      >
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-elc81v-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            info
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-elc81v-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            notice
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-elc81v-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            beta
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-elc81v-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            negative
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-elc81v-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            positive
          </span>
        </div>
      </td>
      <td
        class="css-ebdvcw"
      >
        <div
          class="css-ov1ktg"
        >
          <div
            class="css-1ago99h"
          >
            <div
              class="css-bikrfl"
            >
              <div
                class="css-1asg04x"
              >
                356
              </div>
            </div>
          </div>
        </div>
      </td>
    </tr>
    <tr
      class="css-1lvje81"
    >
      <td
        class="cell-component css-4zu111"
      >
        <div
          class="css-9lvjku"
        >
          <div
            class="css-1ft6wgl"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-25m3b5-MuiSvgIcon-root"
              data-file-name="IconDownloadLarge"
              data-testid="IconDownloadLarge"
              fillcontrast="white"
              focusable="false"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            />
          </div>
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-elc81v-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Downloadable Content
            </span>
          </div>
        </div>
      </td>
      <td
        class="css-p6nomj"
      >
        <div
          class="css-ov1ktg"
        >
          <div
            class="css-1ago99h"
          >
            <div
              class="css-7m17a9"
            >
              <div
                class="css-1asg04x"
              >
                Primary Text
              </div>
            </div>
            <span
              class="css-1xqvcqn"
            >
              Secondary Text
            </span>
            <span
              class="css-1xqvcqn"
            >
              Tertiary Text
            </span>
          </div>
        </div>
      </td>
      <td
        class="css-p6nomj"
      >
        <div
          class="css-ov1ktg"
        >
          <div
            class="css-1ago99h"
          >
            <div
              class="css-7m17a9"
            >
              <div
                class="css-1asg04x"
              >
                Primary Text
              </div>
            </div>
            <span
              class="css-1xqvcqn"
            >
              Secondary Text
            </span>
          </div>
        </div>
      </td>
      <td
        class="cell-component css-4umyx6"
      >
        <div
          aria-labelledby="demo-radio-buttons-group-label"
          class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-c4vnp1-MuiFormControlLabel-root"
          >
            <span
              caption="Caption I"
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary css-4m8g7g-MuiButtonBase-root-MuiRadio-root"
              intent="default"
              label="Option I"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="radio-buttons-group"
                type="radio"
                value="1"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-file-name="IconRadioChecked"
                data-testid="IconRadioChecked"
                fillcontrast="white"
                focusable="false"
                viewBox="0 0 16 16"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              <span
                class="css-1ddu512"
              >
                <span
                  class="css-1dr5e88"
                >
                  Option I
                </span>
                <span
                  class="css-1mg4zid"
                >
                  Caption I
                </span>
              </span>
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-c4vnp1-MuiFormControlLabel-root"
          >
            <span
              caption="Caption II"
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-4m8g7g-MuiButtonBase-root-MuiRadio-root"
              intent="default"
              label="Option II"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="radio-buttons-group"
                type="radio"
                value="2"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-file-name="IconRadioUnchecked"
                data-testid="IconRadioUnchecked"
                fillcontrast="white"
                focusable="false"
                viewBox="0 0 16 16"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              <span
                class="css-1ddu512"
              >
                <span
                  class="css-1dr5e88"
                >
                  Option II
                </span>
                <span
                  class="css-1mg4zid"
                >
                  Caption II
                </span>
              </span>
            </span>
          </label>
        </div>
      </td>
      <td
        class="css-ebdvcw"
      >
        <div
          class="css-ov1ktg"
        >
          <div
            class="css-1ago99h"
          >
            <div
              class="css-bikrfl"
            >
              <div
                class="css-1asg04x"
              >
                1,234
              </div>
            </div>
            <span
              class="css-1xqvcqn"
            >
              2,344,000
            </span>
            <span
              class="css-1xqvcqn"
            >
              12.5%
            </span>
          </div>
        </div>
      </td>
    </tr>
    <tr
      class=" css-szib4e"
      data-mui-internal-clone-element="true"
      disabled=""
    >
      <td
        class="cell-component css-4zu111"
      >
        <div
          class="css-9lvjku"
        >
          <div
            class="css-1ft6wgl"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-25m3b5-MuiSvgIcon-root"
              data-file-name="IconBacteriaLarge"
              data-testid="IconBacteriaLarge"
              fillcontrast="white"
              focusable="false"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            />
          </div>
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-elc81v-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Disease
            </span>
          </div>
        </div>
      </td>
      <td
        class="css-1h7dd7c"
      >
        <div
          class="css-ov1ktg"
        >
          <div
            class="css-1ago99h"
          >
            <div
              class="css-7m17a9"
            >
              <div
                class="css-1asg04x"
              >
                Primary Text
              </div>
            </div>
            <span
              class="css-1xqvcqn"
            >
              Secondary Text
            </span>
            <span
              class="css-1xqvcqn"
            >
              Tertiary Text
            </span>
          </div>
        </div>
      </td>
      <td
        class="css-1h7dd7c"
      >
        <div
          class="css-ov1ktg"
        >
          <div
            class="css-1ago99h"
          >
            <div
              class="css-7m17a9"
            >
              <div
                class="css-1asg04x"
              >
                Primary Text
              </div>
            </div>
          </div>
        </div>
      </td>
      <td
        class="cell-component css-1ssa1s"
      >
        <div
          class="css-1ft6wgl"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-25m3b5-MuiSvgIcon-root"
            data-file-name="IconVirusLarge"
            data-testid="IconVirusLarge"
            fillcontrast="white"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
          />
        </div>
      </td>
      <td
        class="css-ab4j8e"
      >
        <div
          class="css-ov1ktg"
        >
          <div
            class="css-1ago99h"
          >
            <div
              class="css-bikrfl"
            >
              <div
                class="css-1asg04x"
              >
                0.4
              </div>
            </div>
          </div>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
